core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -324,6 +326,8 @@ class TransactionCoordinator(brokerId: Int,
                   logInvalidStateTransitionAndReturnError(transactionalId, txnMetadata.state, txnMarkerResult)
               case Empty =>
                 logInvalidStateTransitionAndReturnError(transactionalId, txnMetadata.state, txnMarkerResult)
+              case Dead =>
```

Comment:
Ditto above, and plus if we cannot skip setting the state to `dead` we can merge these two cases as `Empty | Dead`.